tests/integrations/rediscluster/test_rediscluster.py

```diff
@@ -12,6 +13,11 @@
 
 @pytest.fixture(scope="module", autouse=True)
 def monkeypatch_rediscluster_classes():
+    pipeline_cls = _get_cluster_pipeline_cls(rediscluster)
```

Comment:
just a nit, but why do we need to patch pipeline here? we can use the same `Strict` checking logic above to also decide which of the classes to patch and then just patch `execute/execute_command`?